library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -164,6 +230,47 @@ internal class TransactionPayloadFragment :
         }
     }
 
+    private class FileSaverTask(val fragment: TransactionPayloadFragment) :
+        AsyncTask<Triple<Int, Uri, HttpTransaction>, Unit, Boolean>() {
+
+        @Suppress("NestedBlockDepth")
+        override fun doInBackground(vararg params: Triple<Int, Uri, HttpTransaction>): Boolean {
+            val (type, uri, transaction) = params[0]
+            try {
+                val context = fragment.context ?: return false
+                context.contentResolver.openFileDescriptor(uri, "w")?.use {
+                    FileOutputStream(it.fileDescriptor).use { fos ->
+                        if (type == TYPE_REQUEST) {
+                            transaction.requestBody?.byteInputStream()?.copyTo(fos)
+                        } else if (transaction.responseBody != null) {
+                            transaction.responseBody?.byteInputStream()?.copyTo(fos)
+                        } else {
+                            fos.write(transaction.responseImageData)
+                        }
+                    }
+                }
+            } catch (e: FileNotFoundException) {
+                e.printStackTrace()
+                return false
+            } catch (e: IOException) {
+                e.printStackTrace()
+                return false
+            }
+            return true
+        }
+
+        override fun onPostExecute(result: Boolean) {
+            fragment.fileSaverTask = null
+            fragment.context?.let {
+                if (result) {
+                    Toast.makeText(it, R.string.chucker_file_saved, Toast.LENGTH_SHORT).show()
```

Comment:
What about idea of defining only the resource name in this conditional check and write `Toast` creation only once?
Something like:
```
val toastResource = if (result) {
    R.string.chucker_file_saved
} else {
    R.string.chucker_file_not_saved
}
Toast.makeText(it, toastResource, Toast.LENGTH_SHORT).show()
```